baby-liminal-extension/src/substrate.rs

```diff
@@ -36,45 +40,60 @@ where
     Env: ChainExtensionEnvironment<E, T> + Executor<T>,
     <T as SysConfig>::RuntimeOrigin: From<Option<AccountId32>>,
 {
+    // We assume synthetic lengths for proving and verification keys, necessary to setup benchmarks
+    // with a grid of parameters. The approximate key pair length measured in bytes is capped at
+    // the sum of the caps for the proving (50,000) and verification (10,000) keys.
     #[obce(
         weight(
             expr = r#"{
-                let approx_key_length = env
+                let approx_key_pair_len = env
                     .in_len()
-                    .saturating_sub(size_of::<VerificationKeyIdentifier>() as ByteCount);
+                    .saturating_sub(size_of::<KeyPairIdentifier>() as ByteCount);
 
-                if approx_key_length > 10_000 {
-                    return Ok(RetVal::Converging(BABY_LIMINAL_STORE_KEY_TOO_LONG_KEY));
+                if approx_key_pair_len > 60_000 {
+                    return Ok(RetVal::Converging(BABY_LIMINAL_STORE_KEY_PAIR_TOO_LONG_KEY_PAIR));
                 }
 
-                weight_of_store_key::<T>(approx_key_length)
+                let synthetic_verification_key_len = 1;
```

Comment:
One place worth looking particularly hard at. We want to be able to treat the lengths of both keys as parameters in the benchmarking procedure, hence the use of `synthetic` keys here.